app/src/main/java/org/projectbuendia/client/ui/OdkActivityLauncher.java

```diff
@@ -85,7 +85,37 @@
     private static final Logger LOG = Logger.create();
 
     /**
-     * Fetches all xforms from the server, caches them, and launches ODK using the requested form.
+     * Fetches all xforms from the server and caches them. If any error occurs during fetching,
+     * a failed event is triggered.
+     */
+    public static void fetchAndCacheAllXforms() {
+        new OpenMrsXformsConnection(App.getConnectionDetails()).listXforms(
```

Comment:
Just want to check - this only retrieves the "published" forms, right? Not all of the forms? (there will be a number of forms in OpenMRS that are not published as they were created for other profiles)